### DISCUSSION TITLE
Possible problem with moving_window in Spikemax?

### DISCUSSION THREAD

**Discussion by Michaeljurado42** [lava-nc/lava-dl#127] [Q&A]
Hello and good day to all. 

I was looking at the Spikemax implementation in the case that a moving window is specified. It makes sense to me that with a moving window implementation you would have to have a label for every timestep for every class. The comments seem to support this by specifying a (batch, num_class, time) format. However, I notice that in the MovingWindow confidence function ([here](https://github.com/lava-nc/lava-dl/blob/56099b9c238e383f15a2bd52390c2d8d5cd30b6f/src/lava/lib/dl/slayer/classifier.py)) there is a reshape usage that goes like this: 
`        sliding_rate = self.rate(spike).reshape(
            spike.shape[0], spike.shape[-1]
        )`

This causes an error since the self.rate function produces a 3d tensor which makes the reshape invalid. I made this discussion post and not an issue because of the possibility that I may be using the loss function in the wrong way. If anybody can provide guidance on this I would appreciate it.

**Comment by bamsumit** [comment]
@Michaeljurado42 it looks like an error. It should have been
`sliding_rate = self.rate(spike).reshape( spike.shape[0], -1, spike.shape[-1] )`
Can you test if that works?

**Reply by Michaeljurado42** [reply]
I think that the reshape may be superflous because `self.rate(spike).reshape(spike.shape[0], -1, spike.shape[-1] )` seems to be equivalent to `self.rate(spike)`. Either way, I will make an issue and pull request for this fix and one other problem I recently noticed in the moving_window case soon.

**Comment by bamsumit** [comment]
Thanks @Michaeljurado42 Reshape is needed when you don't have dense layers at the end. In this case, you will have a 5 dimensional tensor. So it's better to enforce it.
